Base/src/main/java/io/deephaven/base/ringbuffer/AggregatingLongRingBuffer.java

```diff
@@ -531,94 +530,128 @@ public long evaluate() {
         return treeStorage[1];
     }
 
-    private static class Range implements Comparable<Range> {
-        int begin;
-        int end;
+    private static class Range {
+        public static final Range zero = new Range(0, 0);
+
+        final int begin;
+        final int end;
 
         public Range(int begin, int end) {
             this.begin = begin;
             this.end = end;
         }
-
-        @Override
-        public int compareTo(Range o) {
-            return Integer.compare(this.begin, o.begin);
-        }
     }
 
     private void fixTree(long r1Head, long r1Tail, long r2Head, long r2Tail, int capacityForThisLevel) {
-//        System.out.println("Calling fixTree()");
+        final Range[] r1s = maybeSplit(r1Head, r1Tail, capacityForThisLevel);
```

Comment:
This would be ugly, but could we avoid allocating arrays by having `r1s`,`r2s` as member variables and having `maybeSplit()` set `Range.ZERO` to `r1s[1]`/`r2s[1]` when there is no overlap?  Then the tests become `if (r1s[1] == Range.ZERO)` instead of `if (r1s.length == 1)` and we save 100M allocations in our tests.